### PR TITLE
Resin and Duroplast

### PR DESCRIPTION
**Pull request by BluSunrize**
It's a lot of stuff! Content wise, it shouldn't warrant much review however.

Important changes to the existing codebase:
- Multiple outputs on the bottling machine
- Bottling machine recipes in the manual

The latter specifically is interesting, because it displays a "fraction" on the bucket item. I enjoy this from a visual and design perspective, because it presents the player with the concept of **buckets** as a unit instead of millibuckets.
However the implementation is maybe a little hacky? Should we use the `lang3.math.Fraction` class here for cleaner lookups or is that library off limits?
Also might be worth moving that coversion to `FluidUtils` instead. We may consider using it for tooltips in the future.

### CODE REVIEW

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java [hunk 1]
```diff
@@ -0,0 +1,199 @@
+/*
+ * BluSunrize
+ * Copyright (c) 2017
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: Wrong copyright year

**Comment by BluSunrize** [comment]
Ready for review

File: src/api/java/blusunrize/immersiveengineering/api/IETags.java [hunk 2]
```diff
@@ -113,6 +115,8 @@
 	public static final TagKey<Fluid> fluidHerbicide = createFluidWrapper(forgeLoc("herbicide"));
 	public static final TagKey<Fluid> fluidRedstoneAcid = createFluidWrapper(forgeLoc("redstone_acid"));
 	public static final TagKey<Fluid> fluidPotion = createFluidWrapper(forgeLoc("potion"));
+	public static final TagKey<Fluid> fluidAcetaldehyde = createFluidWrapper(forgeLoc("ethanal"));
```
Comment (malte0811) [inline src/api/java/blusunrize/immersiveengineering/api/IETags.java]: I feel like that mismatch is going to confuse someone at some point. I can see the reasoning though, and don't have a solution beyond accepting the confusion.

File: src/datagen/java/blusunrize/immersiveengineering/data/IEItemTags.java [hunk 3]
```diff
@@ -73,6 +76,7 @@ protected void addTags()
 		tag(net.minecraft.tags.ItemTags.LECTERN_BOOKS).add(Tools.MANUAL.get());
 		tag(Tags.Items.SEEDS).add(Misc.HEMP_SEEDS.get());
 		tag(Tags.Items.RODS_WOODEN).add(Ingredients.STICK_TREATED.get());
+		tag(ItemTags.PLANKS).add(WoodenDecoration.FIBERBOARD.asItem());
```
Comment (malte0811) [inline src/datagen/java/blusunrize/immersiveengineering/data/IEItemTags.java]: If you add the `planks` tags [here](https://github.com/BluSunrize/ImmersiveEngineering/blob/fd05fa03de87d4a608cd580d9155da5755801944/src/api/java/blusunrize/immersiveengineering/api/IETags.java#L37-L41) we don't have to worry about keeping the contents of the two tags in sync in datagen.

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+					int h = (int)Math.ceil(recipe.output.get().size()/2f);
+					int middle = (int)(h/2f*18);
+
+					ItemStack bucket = recipe.fluidInput.getRandomizedExampleStack(0).getFluid().getBucket().getDefaultInstance();
+					String bucketFraction = getBucketFraction(recipe.fluidInput.getAmount());
+
+					PositionedItemStack[] pIngredients = new PositionedItemStack[recipe.output.get().size()+2];
+					pIngredients[0] = new PositionedItemStack(recipe.input.getItems(), 20, middle);
+					pIngredients[1] = new PositionedItemStack(bucket, 46, middle-8, bucketFraction);
+					final int[] i = {1};
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: IMO a nicer/more explicit way to do this is to use `MutableInt`, makes it clear what it's used for.

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+		if(amount > FluidAttributes.BUCKET_VOLUME)
+		{
+			ret += amount/FluidAttributes.BUCKET_VOLUME;
+			amount = amount%FluidAttributes.BUCKET_VOLUME;
+		}
+		// remaining amount
+		if(amount > 0)
+		{
+			// use fraction symbols where possible
+			if(fractionStrings.containsKey(amount))
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: I personally find the fractions quite hard to read in-game. What do you think about a client config option to skip this branch and just give a decimal representation for everything?

File: src/main/java/blusunrize/immersiveengineering/common/fluids/IEFluid.java [hunk 6]
```diff
@@ -106,11 +106,14 @@ public boolean isSame(@Nonnull Fluid fluidIn)
 		return fluidIn==entry.getStill()||fluidIn==entry.getFlowing();
 	}
 
-	//TODO all copied from water. Maybe make configurable?
 	@Override
 	public int getTickDelay(LevelReader p_205569_1_)
 	{
-		return 5;
+		// viscosity delta to water (1000)
+		int dW = this.getFlowing().getAttributes().getViscosity()-1000;
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/common/fluids/IEFluid.java]: Why not use grab the viscosity for water straight from the fluid attributes for `Fluids.WATER`?

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/jei/bottlingmachine/BottlingMachineRecipeCategory.java [hunk 7]
```diff
@@ -47,9 +47,11 @@ public void setRecipe(IRecipeLayoutBuilder builder, BottlingMachineRecipe recipe
 		builder.addSlot(RecipeIngredientRole.INPUT, 1, 13)
 				.addItemStacks(Arrays.asList(recipe.input.getItems()))
 				.setBackground(JEIHelper.slotDrawable, -1, -1);
-		builder.addSlot(RecipeIngredientRole.OUTPUT, 101, 13)
-				.addItemStack(recipe.output.get())
-				.setBackground(JEIHelper.slotDrawable, -1, -1);
+		int i = 0;
+		for(ItemStack out : recipe.output.get())
+			builder.addSlot(RecipeIngredientRole.OUTPUT, 101, 13+(i++)*18)
```
Comment (malte0811) [inline src/main/java/blusunrize/immersiveengineering/common/util/compat/jei/bottlingmachine/BottlingMachineRecipeCategory.java]: IMO a traditional indexed for loop would be more readable here, using the return value of pre/post-increment is always a bit confusing.

**Review by malte0811 — CHANGES_REQUESTED** [review]

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java (hunk 1 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: Noticed that when working on the manual restructuring, but I can fix it in here too.

File: src/api/java/blusunrize/immersiveengineering/api/IETags.java (hunk 2 above)
Comment (BluSunrize) [reply src/api/java/blusunrize/immersiveengineering/api/IETags.java]: I mean, we can make the tag "acetaldehyde" too. That might be just as confusing, but at least it's consistent. =P

File: src/datagen/java/blusunrize/immersiveengineering/data/IEItemTags.java (hunk 3 above)
Comment (BluSunrize) [reply src/datagen/java/blusunrize/immersiveengineering/data/IEItemTags.java]: okay!

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java (hunk 4 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: I wish IntelliJ suggested me that!!

File: src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java (hunk 5 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/client/manual/ManualElementBottling.java]: Hrm, I suppose.

I really like their visuals myself, but you have a point.

File: src/main/java/blusunrize/immersiveengineering/common/util/compat/jei/bottlingmachine/BottlingMachineRecipeCategory.java (hunk 7 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/common/util/compat/jei/bottlingmachine/BottlingMachineRecipeCategory.java]: Fair

File: src/main/java/blusunrize/immersiveengineering/common/fluids/IEFluid.java (hunk 6 above)
Comment (BluSunrize) [reply src/main/java/blusunrize/immersiveengineering/common/fluids/IEFluid.java]: Hadn't thought about it tbh. >_>

**Comment by BluSunrize** [comment]
Changes made, ready for re-review

**Review by malte0811 — APPROVED** [review]

**Pull request merged**